packages/core/src/factory.ts

```diff
@@ -100,6 +76,21 @@ export function factory<Resource, Input, Output, Config extends IPrismConfig>(
             },
           };
         }),
+        TaskEither.mapLeft(error => {
+          if (components.validateInput) {
+            const { message, name, status } = error as ProblemJsonError;
+            // let's just stack it on the inputValidations
+            // when someone simply wants to hit an URL, don't block them
+            inputValidations.push({
```

Comment:
let' maybe pass `inputValidations` as `output` and `resource` are

Reply:
Hmm I fail to understand what you mean here, can you elaborate?

Reply:
`inputValidations` should be composed inside the chain, and passed down through the chain (ie. like `output`). It shouldn't be a global that's being mutated - `const inputValidations: IPrismDiagnostic[] = [];`. LMK if this explains :)

Reply:
Although it's what I'd like to do, this is really complicated at this stage and I'm going to have to pass on.

The problem is that in order to accumulate all these errors the validators should be kind of refactored and the returned types unified, and we're really not at this stage right now.

I've tried this in the past and the result is basically TypeScript screaming type errors everywhere, and their resolution is not easy at all :(

Reply:
I'm not that hardcore myself, but if TypeScript becomes a problem, you can always try PureScript ;)